input/otlp/traces.go

```diff
@@ -1056,3 +1057,55 @@ func schemeDefaultPort(scheme string) int {
 	}
 	return 0
 }
+
+// parses traceparent header, which is expected to be in the W3C Trace-Context
+// and searches for the p-value as specified in
+//
+//	https://opentelemetry.io/docs/reference/specification/trace/tracestate-probability-sampling/#p-value
+//
+// to calculate the adjusted count (i.e. representative count)
+func getRepresentativeCountFromTracestateHeader(tracestace string) float64 {
+	var p float64 = 0.0
+
+	otValue := getValueForKeyInString(tracestace, "ot", ',', '=')
+
+	if otValue != "" {
+		pValue := getValueForKeyInString(otValue, "p", ';', ':')
+
+		if pValue != "" {
+			p, _ = strconv.ParseFloat(pValue, 64)
```

Comment:
The spec states:

> Both fields are propagated via the OpenTelemetry tracestate under the ot vendor tag using the rules for [tracestate handling](https://opentelemetry.io/docs/reference/specification/trace/tracestate-handling/). Both fields are represented as unsigned decimal integers requiring at most 6 bits of information.

Should we use `strconv.ParseUint(pValue, 10, 6)` then? This would ignore spurious tracestates, as the spec says we should.